src/core/CMakeLists.txt

```diff
@@ -1727,6 +1727,9 @@ set(IMAGE_RCCS ../../images/images.qrc)
 # qgis_core library
 add_library(qgis_core ${LIBRARY_TYPE} ${QGIS_CORE_SRCS} ${QGIS_CORE_HDRS} ${QGIS_CORE_PRIVATE_HDRS} ${IMAGE_RCCS})
 
+# require c++17
+target_compile_features(qgis_core PRIVATE cxx_std_17)
```

Comment:
I think that making it PUBLIC here will avoid specifying the CXX version in each module.

Reply:
Ok, I tried that but it's not possible. If we set it to PUBLIC then the sip build targets inherit this requirement, yet those need to be capped at c++14 due to their use of the deprecated throw(...) annotation (which is used by sip to raise the correct exception class in python, instead of a generic "exception" class). And it's not possible to downgrade the requirement to 14 for the sip targets if it's already set higher by core.